### PR TITLE
storage: add MakeEngineIterator and {Put,Clear}EngineKey

### PR DESCRIPTION
**Pull request by sumeerbhola**
- The EngineIterator interface is implemented by
  pebbleIterator. There are changes to method names
  in EngineIterator since Go does not allow method
  overloading.
- Callers that need to work with EngineKeys, since they will
  eventually need to see separated lock table keys in a
  non-interleaved manner, are switched to using the new
  interfaces.
- ReplicaDataIterator is split into ReplicaMVCCDataIterator
  and ReplicaEngineDataIterator. Only the gcIterator uses
  the former.
- Add PrintEngineKeyValue.
- There was a bug in EngineKey encoding that caused it to
  diverge from MVCCKey. This has been fixed and tests
  added.
- spanset wrappers implement these EngineKey methods for use
   in race tests.

Informs cockroachdb#41720

Release note: None

### CODE REVIEW

**Review by sumeerbhola — COMMENTED** [review]
The first commit is identical to what you have already reviewed. The second commit adds the missing functionality. Ignore the change to `RaceEnabled` -- I'll revert it after CI has run all the tests (I verified the broken tests, but running everything on my machine is too slow).

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-:-MLPCbWe57-fL1bOJo_T:b-hw95hz)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten and @petermattis)

**Review by sumeerbhola — COMMENTED** [review]
@nvanbenschoten 
there are two test failures
- TestLogic/fakedist/uuid: I am unable to reproduce locally, and based on breakpoints it is never using the EngineKey paths in the wrappers that are implemented in spanset/batch.go. So I'm going to ignore this for now.
- TestTenantLogic/3node-tenant/aggregate/other: It doesn't consistently timeout on my machine. Sometimes it completes in ~200s. I was speculating that there is some tenant tear-down activity that maybe using the EngineKey paths that the code is waiting for to complete, but I can't quite figure out what is going on with the `useTenant` path in `logic.go`. Do you have any suggestions, given you know all about the tenant stuff? 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-:-MLTJKebBYxVWvaqMRbK:b4uk6ea)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten and @petermattis)

**Review by petermattis — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-:-MLYkht0970eEFMKdM1X:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten, @petermattis, and @sumeerbhola)

___
*[pkg/kv/kvserver/spanset/batch.go, line 279 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-MLYkUFn5ksGFjfXZgAE:-MLYkUFn5ksGFjfXZgAF:b-8mt6tv) ([raw file](https://github.com/cockroachdb/cockroach/blob/b744bcba639a5a719000d3d7b99f27232b2760f8/pkg/kv/kvserver/spanset/batch.go#L279)):*
> ```Go
> 	if err = i.spans.CheckAllowed(SpanReadOnly, roachpb.Span{Key: key.Key}); err != nil {
> 		// Invalid, but no error.
> 		//nolint:returnerrcheck
> ```

Nit: I think this can be `// nolint:returnerrcheck` (i.e. a space between the `//` and the `nolint:` directive is ok). I've also seen this placed on the return itself:

```
return false, err // nolint:returnerrcheck
```
___
*[pkg/util/race_off.go, line 16 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-MLYkePH7Qbkm1bXPcM-:-MLYkePH7Qbkm1bXPcM0:bfcyzc) ([raw file](https://github.com/cockroachdb/cockroach/blob/b744bcba639a5a719000d3d7b99f27232b2760f8/pkg/util/race_off.go#L16)):*
> ```Go
> 
> // RaceEnabled is true if CockroachDB was built with the race build tag.
> const RaceEnabled = true
> ```

Reminder to revert this before merging.

**Review by nvb — APPROVED** [review]
:lgtm: 

At some point, we'll want to revisit this `spanset` package. The timestamp handling in it is mostly nonsensical and unexercised, and that's added a lot of unnecessary complexity.

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-:-MLeZW3D5aVCzjqoWrVJ:b-wshray)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @nvanbenschoten and @sumeerbhola)

**Review by sumeerbhola — COMMENTED** [review]
> TestTenantLogic/3node-tenant/aggregate/other: It doesn't consistently timeout on my machine. Sometimes it completes in ~200s. I was speculating that there is some tenant tear-down activity that maybe using the EngineKey paths that the code is waiting for to complete, but I can't quite figure out what is going on with the useTenant path in logic.go. Do you have any suggestions, given you know all about the tenant stuff?

@nvanbenschoten  any input on this? the timeout happened in https://teamcity.cockroachdb.com/viewLog.html?buildId=2421381&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_Test
At this point I can't reproduce it at all on my machine.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-:-MLTQBuq7DLsu3aFUypU:bpqj77d)* status: :shipit: complete! 2 of 0 LGTMs obtained (waiting on @sumeerbhola)

**Comment by nvanbenschoten** [comment]
I don't have any idea why the tenant test would be different, but I believe the stacktrace of the currently executing statement is:
```
goroutine 110552 [runnable]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.contains(0xc009230fe0, 0x9, 0x10, 0x0, 0x0, 0x0, 0xc001c97370, 0x9, 0x10, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/spanset.go:319 +0x1d8
github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.(*SpanSet).checkAllowed(0xc0054ee240, 0x1, 0xc001c97370, 0x9, 0x10, 0x0, 0x0, 0x0, 0x4f2a118, 0xc00a4b27b4, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/spanset.go:306 +0x1bf
github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.(*SpanSet).CheckAllowed(0xc0054ee240, 0x1, 0xc001c97370, 0x9, 0x10, 0x0, 0x0, 0x0, 0xa, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/spanset.go:243 +0x65
github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.spanSetWriter.checkAllowed(0x7fc761c8cb58, 0xc00434ef00, 0xc0054ee240, 0x1, 0x0, 0x0, 0xc001c97370, 0x9, 0x10, 0xd, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/batch.go:442 +0xc9
github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset.spanSetWriter.PutIntent(0x7fc761c8cb58, 0xc00434ef00, 0xc0054ee240, 0x1, 0x0, 0x0, 0xc001c97370, 0x9, 0x10, 0xc00a4b2780, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/spanset/batch.go:553 +0x6d
github.com/cockroachdb/cockroach/pkg/storage.(*putBuffer).putMeta(0xc003c72600, 0x7fc761c8ccc8, 0xc0088db560, 0xc001c97370, 0x9, 0x10, 0x0, 0x0, 0xc003c72668, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:1214 +0x2a5
github.com/cockroachdb/cockroach/pkg/storage.mvccPutInternal(0x5a6b120, 0xc00410d950, 0x7fc761c8ccc8, 0xc0088db560, 0x0, 0x0, 0xc00b4c4700, 0xc001c97370, 0x9, 0x10, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:1870 +0x120e
github.com/cockroachdb/cockroach/pkg/storage.mvccPutUsingIter(0x5a6b120, 0xc00410d950, 0x7fc761c8ccc8, 0xc0088db560, 0x0, 0x0, 0xc00b4c4700, 0xc001c97370, 0x9, 0x10, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:1326 +0x17b
github.com/cockroachdb/cockroach/pkg/storage.mvccConditionalPutUsingIter(0x5a6b120, 0xc00410d950, 0x7fc761c8ccc8, 0xc0088db560, 0x0, 0x0, 0xc00b4c4700, 0xc001c97370, 0x9, 0x10, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:2047 +0x1b1
github.com/cockroachdb/cockroach/pkg/storage.MVCCBlindConditionalPut(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:2032
github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.ConditionalPut(0x5a6b120, 0xc00410d950, 0x7fc761c8cbf0, 0xc0088db560, 0x5bfa020, 0xc000a3e9a0, 0x1644c1489c8e28cd, 0x0, 0x100000001, 0x1, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_conditional_put.go:57 +0x276
github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateCommand(0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0x12c3, 0x7fc761c8cbf0, 0xc0088db560, 0x5bfa020, 0xc000a3e9a0, 0xc00b4c4700, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_evaluate.go:468 +0x226
github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateBatch(0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0x7fc761c8cbf0, 0xc0088db560, 0x5bfa020, 0xc000a3e9a0, 0xc00b4c4700, 0xc0012e5050, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_evaluate.go:244 +0x38a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatchWrapper(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0x5bfa020, 0xc000a3e9a0, 0xc00b4c4700, 0xc0012e5050, 0xc0054ee240, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:601 +0x138
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatchWithServersideRefreshes(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0x5bfa020, 0xc000a3e9a0, 0xc00b4c4700, 0xc0012e5050, 0xc0054ee240, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:570 +0x13a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatch(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0xc0012e5050, 0xc0054ee240, 0x5bfd760, 0xc0073b9e60, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:389 +0x212
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateProposal(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0xc0012e5050, 0xc0054ee240, 0x0, 0xc005e1d700, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_proposal.go:738 +0x108
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).requestToProposal(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc008c81708, 0x8, 0xc0012e5050, 0xc0054ee240, 0x0, 0xc005e1d888)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_proposal.go:865 +0x8e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evalAndPropose(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc0012e5050, 0xc004026230, 0xc005e1e3c0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_raft.go:74 +0xd3
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeWriteBatch(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc0012e5050, 0x0, 0x0, 0x0, 0x100000001, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:135 +0x778
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeBatchWithConcurrencyRetries(0xc004cf9800, 0x5a6b120, 0xc00410d950, 0xc0012e5050, 0x4f29bf0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:354 +0x3f5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithRangeID(0xc004cf9800, 0x5a6b120, 0xc00410d920, 0x24, 0xc0012e5050, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:99 +0x5b9
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send(0xc004cf9800, 0x5a6b120, 0xc00410d920, 0x1644c1489c8e28cd, 0x0, 0x100000001, 0x1, 0x0, 0x24, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:36 +0x93
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send(0xc00443f100, 0x5a6b120, 0xc00410d8f0, 0x1644c1489c8e28cd, 0x0, 0x100000001, 0x1, 0x0, 0x24, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:194 +0x5af
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).Send(0xc003d6e7e0, 0x5a6b120, 0xc00410d8f0, 0x0, 0x0, 0x100000001, 0x1, 0x0, 0x24, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/stores.go:177 +0x108
github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal.func1(0x5a6b120, 0xc00410d8c0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:898 +0x220
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0018a9a70, 0x5a6b120, 0xc00410d8c0, 0x494535a, 0x10, 0xc000d8b5e0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:325 +0x140
github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc004f8e100, 0x5a6b120, 0xc00410d8c0, 0xc0039a2e10, 0xc00410d8c0, 0xc00004e000, 0x43c1ba0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:886 +0x197
github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc004f8e100, 0x5a6b120, 0xc00410d860, 0xc0039a2e10, 0xc004f8e100, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:924 +0x9f
github.com/cockroachdb/cockroach/pkg/roachpb._Internal_Batch_Handler.func1(0x5a6b120, 0xc00410d860, 0x48ed7e0, 0xc0039a2e10, 0x0, 0x0, 0xc000d8b7b8, 0x17a3e29)
	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:10694 +0x86
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1(0x5a6b120, 0xc00410d860, 0x48ed7e0, 0xc0039a2e10, 0xc00902eb60, 0xc00902eb80, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:125 +0x69e
google.golang.org/grpc.getChainUnaryHandler.func1(0x5a6b120, 0xc00410d860, 0x48ed7e0, 0xc0039a2e10, 0xc000d8b948, 0xc784a8, 0x460f3e0, 0xc003996180)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1(0x5a6b120, 0xc00410d860, 0x48ed7e0, 0xc0039a2e10, 0xc00902eb60, 0xc003996180, 0xc003996180, 0x0, 0x0, 0xc007888b50)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176 +0xa8
google.golang.org/grpc.getChainUnaryHandler.func1(0x5a6b120, 0xc00410d860, 0x48ed7e0, 0xc0039a2e10, 0xc0039a2e10, 0x0, 0x0, 0xc0054ee1e0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor(0x5a6b120, 0xc00410d860, 0x48ed7e0, 0xc0039a2e10, 0xc00902eb60, 0xc003996140, 0xc000d8ba80, 0xc784a8, 0x460f3e0, 0xc003996140)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:60 +0x8a
google.golang.org/grpc.chainUnaryServerInterceptors.func1(0x5a6b120, 0xc00410d800, 0x48ed7e0, 0xc0039a2e10, 0xc00902eb60, 0xc00902eb80, 0xc0083fab58, 0x54def8, 0x462c000, 0xc00410d800)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:907 +0xca
github.com/cockroachdb/cockroach/pkg/roachpb._Internal_Batch_Handler(0x48a27a0, 0xc004f8e100, 0x5a6b120, 0xc00410d800, 0xc0054ee180, 0xc00232eb60, 0x5a6b120, 0xc00410d800, 0xc00c540000, 0xad604)
	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:10696 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc009f8c1a0, 0x5aedf40, 0xc005396600, 0xc007d8c000, 0xc002c8fd70, 0x82c05e0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1082 +0x4fd
google.golang.org/grpc.(*Server).handleStream(0xc009f8c1a0, 0x5aedf40, 0xc005396600, 0xc007d8c000, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1405 +0xd25
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0049a3950, 0xc009f8c1a0, 0x5aedf40, 0xc005396600, 0xc007d8c000)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:746 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:744 +0xa1
```
It's interesting how close this is to some of the code touched in this PR. The goroutine is runnable, so it's not blocked on anything, but maybe it's stuck in an infinite loop? Are all of the reproductions stuck here?

**Comment by sumeerbhola** [comment]
Thanks, I missed that. I just checked the other ones and the `TestLogic` ones that timed out are not stuck there.
The one other 3node-tenant timeout is also not stuck there https://teamcity.cockroachdb.com/viewLog.html?buildId=2427651&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_Test#testNameId6773925815027051542

The code for the check doesn't suggest a possibility of an infinite loop. Do you think very low cpu on one of the test machines could explain this, and can that happen (I don't know how our CI test machines are provisioned)? `TestTenantLogic/3node-tenant/aggregate/other`, which is the one that fails, is quite expensive and takes 200+ seconds on my machine.

**Comment by nvanbenschoten** [comment]
> Do you think very low cpu on one of the test machines could explain this, and can that happen (I don't know how our CI test machines are provisioned)? TestTenantLogic/3node-tenant/aggregate/other, which is the one that fails, is quite expensive and takes 200+ seconds on my machine.

It's possible. Does the test take that long on your machine when RaceEnabled is set to false? Perhaps we're doing something very expensive under race.

**Comment by sumeerbhola** [comment]
> Does the test take that long on your machine when RaceEnabled is set to false? Perhaps we're doing something very expensive under race.

~20s when RaceEnabled is false.

**Comment by sumeerbhola** [comment]
TFTRs!

CI is green, now that I have set RaceEnabled=false. I'm out of specific ideas to pursue, but I'm somewhat wary of breaking the build again.

**Comment by nvanbenschoten** [comment]
Have you tried running this test on `master` with `RaceEnabled=true`? I guess the main thing to confirm here is that this change wasn't responsible for the timeout. If it takes 200+ seconds on master as well, I'd feel ok merging this PR.

**Review by sumeerbhola — COMMENTED** [review]
There are logictest timeouts on master too with `RaceEnabled=true`
https://teamcity.cockroachdb.com/viewLog.html?buildId=2428346&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_Test#testNameId3612422901405828092

I've removed the known failures in the optimizer (that are being fixed elsewhere) from https://github.com/cockroachdb/cockroach/pull/56442 to see what happens.

I'll then proceed with merging this PR.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-:-MLiAAcY4rlJ_VADueKe:b-1xpusp)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale) (waiting on @nvanbenschoten and @sumeerbhola)

___
*[pkg/kv/kvserver/spanset/batch.go, line 279 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/56341#-MLYkUFn5ksGFjfXZgAE:-MLhQgeB3wr0oHIAxiCX:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/b744bcba639a5a719000d3d7b99f27232b2760f8/pkg/kv/kvserver/spanset/batch.go#L279)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Nit: I think this can be `// nolint:returnerrcheck` (i.e. a space between the `//` and the `nolint:` directive is ok). I've also seen this placed on the return itself:

```
return false, err // nolint:returnerrcheck
```
</blockquote></details>

Done

**Comment by sumeerbhola** [comment]
another master test timeout https://teamcity.cockroachdb.com/viewLog.html?buildId=2428442&tab=buildResultsDiv&buildTypeId=Cockroach_UnitTests_Test

**Pull request merged**
